libvast/vast/data.hpp

```diff
@@ -356,6 +357,48 @@ struct sum_type_access<vast::data> : default_sum_type_access<vast::data> {};
 
 } // namespace caf
 
+namespace vast {
+
+// -- extraction -------------------------------------------------------
+
+/// Tries to retrieve the value associated to `name` from `xs`.
+const data* get_if(const record* xs, std::string_view name);
+
+/// Tries to retrieve the value associated to `name` from `xs`.
+template <class T>
+caf::expected<T> get_if(const record* xs, std::string_view name) {
+  auto data = get_if(xs, name);
+  if (!data)
+    return make_error(ec::convert_error, "record has no", std::string{name},
+                      "field");
+  auto value = caf::get_if<T>(data);
+  if (!value)
+    return make_error(ec::convert_error, std::string{name},
+                      "has an unexpected type:", *data);
+  return *value;
+}
+
+template <class T>
+caf::expected<std::vector<T>> get_vec(const record* xs, std::string_view name) {
+  auto data = get_if(xs, name);
+  std::vector<T> result;
+  if (!data)
+    return result;
```

Comment:
Whether we want to return the empty vector or an error here should probably be governed by `policy::optional` and `policy::required`. Opinions?

Reply:
I would not go there, per my above comment.